src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -37,10 +37,31 @@ public static bool RequiresReflectionMethodBodyScannerForCallSite (LinkContext c
 			if (methodDefinition == null)
 				return false;
 
-			return
-				GetIntrinsicIdForMethod (methodDefinition) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel ||
+			bool isPInvokeImpl = methodDefinition.IsPInvokeImpl;
+			if (isPInvokeImpl) {
+				// HACK - we don't want to scan C++/CLI assemblies
```

Comment:
Why are we avoiding C++/CLI?

Reply:
Primarily as a convenience in the initial support. Our existing analysis of the method body instructions seem inadequate from the few tests we had (IL scanning errors) and given native code interaction to be very common in C++/CLI, I avoided adding support in this iteration.

Reply:
If we are concerned the method body scanner could crash, should we also block scanning for reflection patterns then?

C++/CLI can still do stuff that would trigger the method body scanner (call into reflection intrinsics, or RUC code). If we are afraid we would crash, we need to do something about that. All those crashes are just bugs - there's nothing fundamentally hard in scanning the IL bodies of C++/CLI assemblies.

Reply:
If we think C++/CLI is broken, OK - let's fix it, but I don't see a reason to try to "make up for it" in one specific area. I personally would remove any C++/CLI specific handling here - unless we need new logic to correctly detect PInvokes in C++/CLI (per the other discussion).